### ISSUE TITLE
Sending HTML pages instead of plain text

### ISSUE THREAD

**Issue by SaeidYazdani** [etr/libhttpserver#234]
Sorry if this is not the right place. can you please tell me what should I change/do to send a HTML page instead of plain text? so that the browser renders it as heml? I am trying to to the hello world example but it just gets sent as plain text:

```
class hello_world_resource : public http_resource {
public:
    const std::shared_ptr<http_response> render(const http_request&) {
        return std::shared_ptr<http_response>(
                new string_response("<html><body><h1>HI THERE</h1></body></html>"));
    }
};
```

**Comment by etr** [comment]
Sorry for the delay on this answer.

In order for your browser to interpret the string returned as HTML, you will have to pass a content-type header set to the "text/html"

Specifically, in libhttpserver, you can pass that to the constructor of string response:

    new string_response("<html><body><h1>HI THERE</h1></body></html>", 200, "text/html")

Hope this helps.

**Comment by angstyloop** [comment]
This means you can serve HTML from a file using string_response, with just a few more lines of code. This is almost definitely the totally wrong way to serve a static HTML web page with libhttpserver, but hey, you're serving HTML with like 20 lines of C++. Just enjoy the moment.

```
#include <httpserver.hpp>
#include <fstream>
#include <iostream>

class hello_world_resource : public httpserver::http_resource {
 public:
     std::shared_ptr<httpserver::http_response> render(const httpserver::http_request&) {
         std::ifstream f("./examples/test_content");
         std::stringstream s;
         s << f.rdbuf();
         auto res = new httpserver::string_response(s.str(), 200, "text/html");
         return std::shared_ptr<httpserver::http_response>(res);
     }
};

int main() {
    httpserver::webserver ws = httpserver::create_webserver(8080);

    hello_world_resource hwr;
    ws.register_resource("/hello", &hwr);
    ws.start(true);

    return 0;
}

```

Your `test_content` file can be any valid HTML - httpserver::string_response is just sending the string you pass to it - it's up to the web browser to parse it. Having said that, throw some JavaScript in there if you want.

```
<html>
<head>  
        <title>no seriously, libhttpserver is pretty cool</title>
</head>
<body>
        <h1>yep, libhttpserver is pretty cool</h1>
        <script>alert('so I heard libhttpserver is pretty cool...')</script>
</body>
</html>
```

I'm just editing the examples ( the .cpp files ) in `libhttpserver/examples/` and rebuilding with my changes ( by calling `make` again in `libhttpserver/build/` ), so the relative path to your `test_content` file may be different than mine.

Note that `make` doesn't pick up changes to `libhttpserver/examples/test_content`, so you'll have to copy it into `libhttpserver/build/examples/` by hand, or just edit it right there in the `build/` directory. There's no need to rebuild after making changes to `test_content` - just make sure you're editing `test_content` in `build/` ( or copying it there after making changes ), or obviously you won't see your changes in your browser.

**Comment by etr** [comment]
If you are starting from a file, you don't really need to read the file yourself and then use a string_response. You can use file_response, that should make your life easier.

**Comment by cschill2020** [comment]
Thanks for this awesome library!  I am testing out various functions, but I am scratching my head a bit for the following case.  
I want to register "/" to serve an html file.  The html file also includes a css and a javascript file that need to load.
So the html file has a line:
<link rel="stylesheet" type="text/css" href="mystyle.css">

In this case the browser is trying to get http://serveraddress:port/mystyle.css

I got this to work by registering a "/mystyle.css" endpoint.  But what I would like to do instead is have the code in one resource:
if (filename empty) return myhtml.html
else return filename

Basically, I am wondering if it is possible to create one endpoint that is handling all the "server:port/file.xxx" requests.

I am a bit new to http server in c++.  All my main classes for handling the requests are in c++ and I will have a bunch of js,html,png files needed to serve the webapp.
